src/raw/windows.rs

```diff
@@ -6,137 +6,94 @@ use std::cmp;
 use std::ptr;
 use std::sync::{Once, ONCE_INIT};
 
-//WinAPI Start
 mod winapi {
     #![allow(bad_style)]
     #![allow(dead_code)]
 
-    use std::os::raw::*;
-
-    pub type SOCKET = ::std::os::windows::io::RawSocket;
-    pub type DWORD = c_ulong;
-    pub type WORD = c_ushort;
-    pub type GROUP = c_uint;
-    pub type CHAR = c_char;
-    pub type USHORT = c_ushort;
-    pub type ADDRESS_FAMILY = USHORT;
-    pub const INVALID_SOCKET: SOCKET = !0;
-    pub const SOCKET_ERROR: c_int = -1;
-    pub const AF_INET: c_int = 2;
-    pub const AF_INET6: c_int = 23;
-    pub const WSAESHUTDOWN: DWORD = 10058;
-    pub const FD_SETSIZE: usize = 64;
-
-    pub const WSADESCRIPTION_LEN: usize = 256;
-    pub const WSASYS_STATUS_LEN: usize = 128;
-    #[repr(C)] #[derive(Copy)]
-    pub struct WSADATA {
-        pub wVersion: WORD,
-        pub wHighVersion: WORD,
-        #[cfg(target_arch="x86")]
-        pub szDescription: [c_char; WSADESCRIPTION_LEN + 1],
-        #[cfg(target_arch="x86")]
-        pub szSystemStatus: [c_char; WSASYS_STATUS_LEN + 1],
-        pub iMaxSockets: c_ushort,
-        pub iMaxUdpDg: c_ushort,
-        pub lpVendorInfo: *mut c_char,
-        #[cfg(target_arch="x86_64")]
-        pub szDescription: [c_char; WSADESCRIPTION_LEN + 1],
-        #[cfg(target_arch="x86_64")]
-        pub szSystemStatus: [c_char; WSASYS_STATUS_LEN + 1],
-    }
-
-    #[repr(C)] #[derive(Copy)]
-    pub struct FD_SET {
-        pub fd_count: c_uint,
-        pub fd_array: [SOCKET; FD_SETSIZE],
-    }
-
-    impl Clone for FD_SET {
-        fn clone(&self) -> FD_SET { *self }
-    }
-
-    impl Clone for WSADATA {
-        fn clone(&self) -> WSADATA { *self }
-    }
-
-    #[repr(C)] #[derive(Clone, Copy)]
-    pub struct timeval {
-        pub tv_sec: c_long,
-        pub tv_usec: c_long,
-    }
-
-    #[repr(C)]
-    pub struct SOCKADDR_STORAGE_LH {
-        pub ss_family: ADDRESS_FAMILY,
-        pub __ss_pad1: [CHAR; 6],
-        pub __ss_align: i64,
-        pub __ss_pad2: [CHAR; 112],
-    }
-
-    #[repr(C)]
-    #[derive(Copy, Clone)]
-    pub struct in_addr {
-        pub s_addr: [u8; 4],
-    }
-
-    #[repr(C)]
-    #[derive(Copy, Clone)]
-    pub struct in6_addr {
-        pub s6_addr: [u16; 8],
-    }
+    extern crate winapi;
+
+	pub use self::winapi::{
+		ADDRESS_FAMILY,
+		SOCKET,

```

Comment:
I think it would be better to use Rust builtin alias to SOCKET from `std::os:windows::io`